### ISSUE TITLE
eventful.registerReaction adds callback function twice

### ISSUE THREAD

**Issue by quarque2** [DFHack/dfhack#2206]
If you add a callback function to a reaction with registerReaction, the callback function is executed two times for each reaction.

This bug is not related to the recent fix for the missing call_native argument. I tested with both the latest stable version and with the latest build. In the latest build the call_native is available, so I can confirm that the recent bugfix for that problem works as intended, but the callback is executed twice in both versions.

**Comment by warmist** [comment]
@quarque2 my guess is that the reaction has two products and each of them triggers the callback. This was not documented before but is now. Although that is annoying and we should probably make something better.

**Comment by myk002** [comment]
Yeah, I looked through the code too and didn't see anything obvious.
